src/utils/entraServicePrincipal.ts

```diff
@@ -0,0 +1,30 @@
+import { ServicePrincipal } from '@microsoft/microsoft-graph-types';
+import { odata } from './odata.js';
+import { formatting } from './formatting.js';
+import { cli } from '../cli/cli.js';
+
+export const entraServicePrincipal = {
+  async getServicePrincipalIdFromAppId(appId: string): Promise<string> {
```

Comment:
Let's return the entire object instead of just the name, and add a properties option as well.


```suggestion
  async getServicePrincipalFromAppId(appId: string, properties?: string): Promise<string> {
```